Tribe Member section Implementation

Korede612 opened the pull request:
# Pull Request Title
## New view for TRIBE MEMBERS in the right panel of the chat view https://github.com/stakwork/sphinx-mac/issues/365

### Description
This pull request addresses the Create New view for TRIBE MEMBERS in the right panel of the chat view in Issue https://github.com/stakwork/sphinx-mac/issues/365.

### The following are the addressed issues:
- A new option should be added Tribe Members in the 3 dots tribe menu only if you are the admin of the tribe 
- Tapping on this new option a new View Controller should be presented in the right panel

## Changes Made
- API+WalletExtension.swift:
    - Added a method called getContactsForChat to fetch the List of Contacts in a Tribe
- API.swift:
    - Added a typealias for the response callback
- Chat+CoreDataClass.swift:
    - Added a new option to the option list for the admin of the tribe
- StoryboardScenes.swift:
    - Added a static property called tribeMembersViewController for TribeMembersViewController
- MessageOptionsHelper.swift:
    - Added a new case TribeMembers to the enum ChatActionsItem
- GroupContactListItem.xib:
    - Created the custom view of type NSCollectionViewItem using interface builder
- GroupContactListItem.swift:
    - Link the xib, properties and added necessary method
- TribeMembersDataSource.swift:
    - Created the datasource for the collection view with the necessary methods
    - Created a method for making calling the getContactsForChat from the API class to fetch the list of contacts in a Tribe
- NewChatViewController.swift:
    - Changed a method argument from true to false (This makes the right panel adjustable for the Thread View)
- DashboardDetailViewController.swift:
    - Added a method for readjusting the layout of the most current ViewController view frame added to the right panel
- DashboardViewController.swift:
    - Called the method for readjusting frame when the right panel changes frame or readjusted
- TribeMembersViewController.storyboard:
    - Create the ViewController using the interface builder
- TribeMembersViewController.swift:
    - Link the necessary outlets from the interface builder
- MessageOptionsDelegate:
    - Added a method for navigating to the TribeMembersViewController
- Localizable.strings:
    - Added a localizable string for the Tribe View title

Korede612 committed:
Merge pull request #6 from stakwork/test-flight

Test flight

Korede612 committed:
feat: Tribe Member section setup

Korede612 committed:
feat: Tribe Members View Controller added

Korede612 committed:
feat: Tribe Member section in progress

Korede612 committed:
feat: Tribe Member section datasource and cellItem

Korede612 committed:
feat: Tribe Member section 95% completed

Korede612 committed:
feat: Tribe Member section 100% completed

Korede612 committed:
fix: Remove print statement

tomastiminskas commented:
@Korede612 the changes look and work great. Just a few adjustments to make:

1. The tribe members are showing in the wrong order. The should be alphabetically ordered starting from the top but they are showing in reverse
<img width="381" alt="Screenshot 2024-05-02 at 10 25 00 AM" src="https://github.com/stakwork/sphinx-mac/assets/2242281/e306690f-8e0e-429f-8283-c1177384d6da">

3. Initials color is wrong, it should be a grey color (that you can take from iOS project) but it's showing white
4. Let's align the initial label with the back arrow on the header (right now there's a small difference)
5. For members without an image you should show the initials with the user color (you should use `ChatSmallAvatarView` class which contains both an image and a view with initials and a background color). If you add one view of that type on each row, then you can call a method on the class with an optional image, an alias and a color)
6. Finally as you might have seen, iOS has the ability to delete a member from the tribe swiping in the row and then tapping on delete button. We don't want a swipe to delete gesture on Mac, but we can just show the delete icon on the right side of each row (tint color secondaryText) and add the logic when tapping on it. Once a member has been deleted you should reload the entire table since the initials logic will be broken otherwise.

Let me know if the changes make sense

Korede612 committed:
fix: UI fixes and deleting of contact from tribe

Korede612 commented:
@tomastiminskas the changes requested makes sense

- All issues mentioned have been completed
- I added an alert for the deleting functionality in order to be sure it wasn't an accident 
- I added a loader in-case of poor internet
